### PR TITLE
[PT FE] Run GPU with FP32 hint when layer test is FP32

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Run GPU with FP32 hint when layer test is FP32*
 - *Xfail tests unsupported on GPU*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 1]
```diff
@@ -138,7 +138,10 @@ def numpy_to_torch_recursively(x):
                             smodel.inlined_graph, op), f"Operation {op} type doesn't exist in provided graph"
             # OV infer:
             core = Core()
-            compiled = core.compile_model(converted_model, ie_device)
+            config = {}
+            if ie_device == "GPU" and precision == "FP32":
+                config["INFERENCE_PRECISION_HINT"] = Type.f32
```
Comment (ilya-lavrenov) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:143]: let's avoid hardcoded strings. We have pythonic way:
```
    config = {hints.inference_precision: ov.Type.f32}
```

File: tests/layer_tests/pytorch_tests/test_cross.py [hunk 2]
```diff
@@ -54,13 +54,17 @@ def forward_no_dim_no_out(self, x, y):
...
     def test_linalg_cross(self, x_shape, y_shape, dim, out, dtype, ie_device, precision, ir_version):
         self._test(
-            *self.create_model(dim, out), ie_device, precision, ir_version, use_convert_model=True, 
-            kwargs_to_prepare_input={"x_shape":x_shape, "y_shape": y_shape, "out": out, 'dtype': dtype})
+            *self.create_model(dim, out), ie_device, precision, ir_version, use_convert_model=True,
+            kwargs_to_prepare_input={"x_shape": x_shape,
+                                     "y_shape": y_shape,
+                                     "out": out,
+                                     'dtype': dtype},
+            dynamic_shapes=False if ie_device == "GPU" else True)
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_cross.py:67]:
```suggestion
            dynamic_shapes=ie_device != "GPU")
```

File: tests/layer_tests/pytorch_tests/test_cross.py [hunk 3]
```diff
@@ -113,10 +117,14 @@ def forward_no_dim_no_out(self, x, y):
...
-                   use_convert_model=True, 
-                   kwargs_to_prepare_input={"x_shape":x_shape, "y_shape": y_shape, "out": out, "dtype": dtype})
\ No newline at end of file
+        self._test(*self.create_model(dim, out, x_shape), ie_device, precision, ir_version,
+                   use_convert_model=True,
+                   kwargs_to_prepare_input={"x_shape": x_shape,
+                                            "y_shape": y_shape,
+                                            "out": out,
+                                            "dtype": dtype},
+                   dynamic_shapes=False if ie_device == "GPU" else True)
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_cross.py:130]:
```suggestion
                   dynamic_shapes=ie_device != "GPU")
```

File: tests/layer_tests/pytorch_tests/test_deformable_convolution.py [hunk 4]
```diff
@@ -172,6 +172,7 @@ def forward(self, x):
     @pytest.mark.precommit
     @pytest.mark.precommit_fx_backend
     def test_deformable_convolution2d(self, params, bias, mask, ie_device, precision, ir_version):
-        self._test(
-            *self.create_model(**params, bias=bias, mask=mask), ie_device, precision, ir_version, trace_model=True
-        )
+        self._test(*self.create_model(**params, bias=bias, mask=mask),
+                   ie_device, precision, ir_version, trace_model=True,
+                   dynamic_shapes=False if ie_device == "GPU" else True
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_deformable_convolution.py:177]:
```suggestion
                   dynamic_shapes=ie_device != "GPU"
```

File: tests/layer_tests/pytorch_tests/test_roll.py [hunk 5]
```diff
@@ -40,4 +40,5 @@ def forward(self, x):
     @pytest.mark.precommit_torch_export
     @pytest.mark.precommit_fx_backend
     def test_roll(self, shifts, dim, ie_device, precision, ir_version):
-        self._test(*self.create_model(shifts, dim), ie_device, precision, ir_version)
+        self._test(*self.create_model(shifts, dim), ie_device, precision, ir_version,
+                   dynamic_shapes=False if ie_device == "GPU" else True)
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_roll.py:44]:
```suggestion
                   dynamic_shapes=ie_device != "GPU")
```

**Review by eaidova — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**